extensions/smallrye-jwt/runtime/src/main/java/io/quarkus/smallrye/jwt/runtime/auth/JWTCredential.java

```diff
@@ -5,14 +5,15 @@
 import org.jose4j.jwt.consumer.JwtConsumerBuilder;
 import org.jose4j.jwt.consumer.JwtContext;
 
+import io.quarkus.elytron.security.runtime.UndertowTokenCredential;
 import io.smallrye.jwt.auth.principal.JWTAuthContextInfo;
 import io.undertow.security.idm.Credential;
 
 /**
  * This is an implementation of the undertow Credential that wraps the bearer token and configured JWTAuthContextInfo
  * needed for validation of the token.
  */
-public class JWTCredential implements Credential {
+public class JWTCredential implements UndertowTokenCredential {
```

Comment:
Sorry, one more question :-), I'm not sure `Undertow` qualifier should be in the class name, and I hope it will work in the new Quarkus HTTP ecosystem even without Undertow :-)

Reply:
That interface is temporary and will be deleted. All the Undertow related IdentityManager/Credential stuff is going to be replaced very soon. I think it will be a lot easier to merge these changes eagerly though rather than having a long lived topic branch, as it will enable much quicker feedback cycles.

Reply:
Sounds good